packages/block-editor/src/components/rich-text/use-input-rules.js

```diff
@@ -86,9 +88,11 @@ export function useInputRules( props ) {
 			} );
 			const block = transformation.transform( content );
 
-			selectionChange( findSelection( [ block ] ) );
+			selectionChange( ...findSelection( [ block ] ) );
```

Comment:
The reason this fixes the failing test is because we preset selection in the store after replacing the input rule. So when the new block is created, the selection is already correct at the time of the focus event.